### PR TITLE
Android support

### PR DESCRIPTION
**Pull request by TheRawMeatball**
Improves situation on #86

Current status: Occasional panics when resuming, needs #2300.

### CODE REVIEW

**Comment by cart** [comment]
> Switch to using stdout logging on android 

Can you elaborate a bit on what motivated this decision? I'm assuming `android_log-sys` was causing problems? Losing "native android log levels" isn't ideal.

**Comment by TheRawMeatball** [comment]
I'm not sure why, but it didn't work properly for me. And given that stdout is already redirected, a consistent output is more appealing to me - the command `adb logcat -v raw RustStdoutStderr:D *:S` gives exactly what you'd get if you were running the game on the PC.

**Comment by cart** [comment]
I think being able to use UI to toggle logs by their level is way more valuable than exact consistency with pc console logs. The android ides I've used all support this for "native android logs". I also think theres a good chance that we will ultimately have a custom tracing backend for the bevy editor that collects a "structured log format" instead of stdout.

Obviously being able to run android apps is more important than this, but I'm still very much in favor of retaining native logging if possible.

**Comment by TheRawMeatball** [comment]
Hmm, I experimented with it a bit more, and personally I'm still in favor of defaulting android to stdout, since right now getting a clean output from `adb logcat` is not super easy. We should consider that most people may not want to set up a full Android IDE, and want to just quickly debug something. I'll probably put the android logger impl behind a feature flag and default to stdout. Would that be an OK compromise?

**Comment by cart** [comment]
> since right now getting a clean output from adb logcat is not super easy

Can you elaborate a bit on this? I wouldn't call `adb logcat -v raw RustStdoutStderr:D *:S` particularly "easy" either. I've successfully used `adb logcat` with the native logger (by just scrolling to my app logs).

**Comment by TheRawMeatball** [comment]
It's more of an I haven't been able to figure it out yet :sweat_smile:. I tried a bunch of stuff for cleaning up the logs, but there was still random irrelevant logs coming in. Especially for a game, where my debugging strategy tends to be spam the console and use it as a real time display, scrolling isn't an option, and random logs coming in can make the text jittery.

**Comment by cart** [comment]
This stackoverflow thread seems to have a number of options which are approximately as complex as `adb logcat -v raw RustStdoutStderr:D *:S`:

https://stackoverflow.com/questions/6854127/filter-logcat-to-get-only-the-messages-from-my-application-in-android

**Comment by TheRawMeatball** [comment]
These still gave pings from android s vulkan layer :/.

File: deny.toml [hunk 1]
```diff
@@ -70,8 +70,8 @@ skip = [
     { name = "getrandom", version = "0.1" },
     { name = "libm", version = "0.1" },
     { name = "mach", version = "0.2" },
-    { name = "ndk", version = "0.2" },
-    { name = "ndk-glue", version = "0.2" },
+    { name = "ndk", version = "0.3" },
+    { name = "ndk-glue", version = "0.3" },
```
Comment (MinerSebas) [inline deny.toml:74]:
```suggestion
```
This should be removed instead as ndk 0.3 is no longer present multiple times in the dependence tree

**Comment by cart** [comment]
bors try

**Comment by cart** [comment]
(I'm hitting shaderc complication errors on my system and I'm curious to see if bors hits them too)

**Comment by cart** [comment]
Yeah I think the issue is that we need `all(target_arch="armv7", target_os="androideabi")` in cargo.toml instead of `all(target_arch="armv7", target_os="androidabi")`

**Comment by cart** [comment]
Hmm i guess this is a "cart's machine" problem. It is weird that shaderc is getting compiled at all? Shouldn't we be using glsl-to-spirv?

**Comment by TheRawMeatball** [comment]
Maybe we wait until after the renderer rework and don't have to bother with shaderc at all?

**Comment by cart** [comment]
I like this plan :)

**Comment by billyb2** [comment]
Just out of curiosity, when will said rework probably come? (Very excited for Android support in Bevy, and it will make it easily the best Rust game engine imho).

**Comment by cart** [comment]
Hard to say, but ideally within the next month.

**Comment by billyb2** [comment]
Alright, thank you!

**Comment by TheRawMeatball** [comment]
#2482 has done most of what this PR does, closing.

**Pull request closed without merging**